projects/pastanaga-angular/src/lib/controls/textfield/input/input.component.ts

```diff
@@ -25,21 +25,29 @@ import { NativeTextFieldDirective } from '../native-text-field.directive';
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class InputComponent extends NativeTextFieldDirective implements OnChanges, OnInit, AfterViewInit, OnDestroy {
-    @Input() set type(value: TextInputType) {
+    @Input()
+    set type(value: TextInputType) {
         this._type = value || 'text';
         this._updateInputType();
     }
-
     get type() {
         return this._type;
     }
 
+    @Input()
+    set icon(value: string | undefined) {
```

Comment:
it's probably too late...
but I would have preffered a specific component like InputIconComponent
that could have been extracted from the ng-content and rendered wherever we want...